File: medium/number_of_letters_in_a_number_binary.py
import sys
import math

# Fonctions
def decimalToBinary(num):
    return bin(num).replace('0b', "")


def binaryToLetters(binary):
    binary_to_str = {
        "0": "zero",
        "1": "one"
    }
    str_binary = ""
    for digit in binary:
        str_binary += binary_to_str[digit]

    return str_binary

# varibles
suite, n = [int(i) for i in input().split()]

for i in range(n):
    last_number = suite
    suite_binary = decimalToBinary(suite)
    str_binary = binaryToLetters(suite_binary)
    suite = len(str_binary)

    if last_number == suite:
        break

print(suite)

File: medium/number_of_letters_in_a_number_binary.rb
# Functions
def binaryToLetters(binary)
    binary_to_str = { "0" => "zero", "1" => "one" }
    str_binary = ""
    binary = binary.split('')
    binary.each do |digit|
        str_binary += binary_to_str[digit]
    end
    return str_binary
end

# Variables
suite, n = gets.split.map { |x| x.to_i }

for i in 1..n
    last_number = suite
    suite_binary = suite.to_s(2)
    str_binary = binaryToLetters(suite_binary)
    suite = str_binary.length

    if last_number == suite
        break
    end
end

puts suite
